tests/functional/TestErrorHandling/tests/background_cleanup_test.go

```diff
@@ -0,0 +1,182 @@
+	info, err := heketi.OperationsInfo()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, info.InFlight == 0,
+		"expected info.InFlight == 0, got:", info.InFlight)
+	tests.Assert(t, info.Stale == 1,
+		"expected info.Stale == 1, got:", info.InFlight)
+
+	time.Sleep(5 * time.Second)
+	for i := 0; i < 5; i++ {
+		// wait around and additional 5 sec for stale ops to be cleaned up
+		time.Sleep(time.Second)
+		info, err = heketi.OperationsInfo()
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+		if info.Stale == 0 {
+			break
+		}
+	}
+
+	info, err = heketi.OperationsInfo()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, info.InFlight == 0,
+		"expected info.InFlight == 0, got:", info.InFlight)
+	tests.Assert(t, info.Stale == 0,
+		"expected info.Stale == 0, got:", info.Stale)
+}
+
+func TestCleanupPeriodic(t *testing.T) {
+	heketiServer := testutils.NewServerCtlFromEnv("..")
+	origConf := path.Join(heketiServer.ServerDir, heketiServer.ConfPath)
+
+	heketiServer.ConfPath = tests.Tempfile()
+	defer os.Remove(heketiServer.ConfPath)
+
+	glusterCond := tests.Tempfile()
+	f, err := os.Create(glusterCond)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, f.Close() == nil, "expected err == nil, got:", err)
+	defer os.Remove(glusterCond)
+
+	UpdateConfig(origConf, heketiServer.ConfPath, func(c *config.Config) {
+		c.GlusterFS.StartTimeBackgroundCleaner = 2
+		c.GlusterFS.RefreshTimeBackgroundCleaner = 8
+		c.GlusterFS.Executor = "inject/ssh"
+		c.GlusterFS.InjectConfig.CmdInjection.CmdHooks = inj.CmdHooks{
+			inj.CmdHook{
+				Cmd:      "^gluster .*volume .*",
+				CondFile: glusterCond,
+				Reaction: inj.Reaction{
+					Err: "failing a gluster volume command",
+				},
+			},
+		}
+	})
+
+	fullTeardown := func() {
+		CopyFile(origConf, heketiServer.ConfPath)
+		testutils.ServerRestarted(t, heketiServer)
+		testCluster.Teardown(t)
+		testutils.ServerStopped(t, heketiServer)
+	}
+	defer fullTeardown()
+
+	testutils.ServerStarted(t, heketiServer)
+	heketiServer.KeepDB = true
+	testCluster.Setup(t, 3, 3)
+	time.Sleep(2 * time.Second) // ensure we pass the start time
+
+	logger.Info("fail to create a volume w/ failed rollback")
+	volReq := &api.VolumeCreateRequest{}
+	volReq.Size = 5
+	volReq.Durability.Type = api.DurabilityReplicate
+	_, err = heketi.VolumeCreate(volReq)
+	tests.Assert(t, err != nil, "expected err != nil, got:", err)
+
+	info, err := heketi.OperationsInfo()
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, info.InFlight == 0,
+		"expected info.InFlight == 0, got:", info.InFlight)
+	tests.Assert(t, info.Failed == 1,
+		"expected info.Failed == 1, got:", info.Failed)
+
+	logger.Info("removing glusterCond condition file")
+	os.Remove(glusterCond)
+	time.Sleep(time.Second)
+
+	for i := 0; i < 16; i++ {
+		// wait around and additional 5 sec for stale ops to be cleaned up
```

Comment:
copy'n'paste error. we do not need to name exact number of seconds in the comment.
Anyways I would put the comment *above* the for loop. But not blocking the merge for this...